apps/web/src/components/auth/RegisterForm.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+  Anchor,
+  Button,
+  Paper,
+  PasswordInput,
+  Stack,
+  Text,
+  TextInput,
+  Input,
+  Group,
+  Flex,
+  Title,
+  Select,
+} from '@mantine/core';
+import React, { useState } from 'react';
+import { Form } from 'ui';
+import * as z from 'zod';
+
+import { StyledContainer } from './styles';
+
+const schema = z.object({
+  email: z.string().min(1, 'Required'),
+  password: z.string().min(8, 'Minimum Password is 8'),
+});
+
+type RegisterValues = {
+  usertype: string;
+  username: string;
+  firstname: string;
+  lastname: string;
+  email: string;
+  street: string;
+  region: string;
+  zipcode: number;
+  country: string;
+  password: string;
+  confirmpassword: string;
+};
+
+type RegisterFormProps = {
+  onSuccess: () => void;
+};
+
+export const RegisterForm = ({ onSuccess }: RegisterFormProps) => {
+  const [data, setData] = useState([
+    { value: 'consumer', label: 'Consumer' },
+    { value: 'merchant', label: 'Merchant' },
+  ]);
+
+  return (
+    <StyledContainer>
+      <Paper radius="md" p="xl" withBorder>
+        <Form<RegisterValues, typeof schema>
+          onSubmit={async () => {
+            onSuccess();
+          }}
+          schema={schema}
+        >
+          {({ formState }) => (
+            <>
+              <Stack>
+                <Title order={1}>Register</Title>
+                <div style={{ display: 'none' }}>
+                  <Select
+                    value={data.values[0]}
+                    label="Creatable Select"
+                    data={data}
+                    placeholder="Select items"
+                    nothingFound="Nothing found"
+                    searchable
+                  />
```

Comment:
What's the use of this if the display is none?

Reply:
Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.